.github/workflows/publish.yml

```diff
@@ -0,0 +1,98 @@
+name: Publish release
+
+on:
+  workflow_dispatch:
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Get latest existing tag
+      uses: actions-ecosystem/action-get-latest-tag@v1
+      id: get-latest-tag
+      with:
+        semver_only: true
+
+    - name: Set new version
+      uses: paulhatch/semantic-version@v4.0.2
+      id: set-version
+      with:
+        tag_prefix: "v"
+        format: "${major}.${minor}.${patch}"
+        major_pattern: "(MAJOR)"
+        minor_pattern: "(MINOR)"
+
+    - name: Generate changelog since last tag
+      uses: mathiasvr/command-output@v1
+      id: changelog
+      with:
+        run: git log --format="* %s" ${{ steps.get-latest-tag.outputs.tag }}..HEAD
+
+    - name: Log version & changelog
+      run: |
+        echo "Version: $VERSION"
+        echo "Version tag: $VERSION_TAG"
+        echo "Latest tag detected: $LATEST_TAG"
+        echo "Changelog: $CHANGELOG"
+      env:
+        VERSION: ${{ steps.set-version.outputs.version }}
+        VERSION_TAG: ${{ steps.set-version.outputs.version_tag }}
+        LATEST_TAG: ${{ steps.get-latest-tag.outputs.tag }}
+        CHANGELOG: ${{ steps.changelog.outputs.stdout }}
+
+    - name: Prevent empty release
+      if: ${{ steps.changelog.outputs.stdout == '' }}
+      uses: actions/github-script@v3
+      with:
+        script: |
+          core.setFailed("No changes since prior release")
+
+    - name: Update changelog
+      run: |
+        (head -8 CHANGELOG.md && echo "## $VERSION" && date "+_%e %B %Y_" && echo "" && echo "$CHANGELOG" && tail -n +9 CHANGELOG.md) > CHANGELOG.new.md
+        mv CHANGELOG.new.md CHANGELOG.md
+      env:
+        VERSION: ${{ steps.set-version.outputs.version }}
+        CHANGELOG: ${{ steps.changelog.outputs.stdout }}
+
+    - name: Install Rust toolchain
+      uses: actions-rs/toolchain@v1
+      with:
+        toolchain: stable
+
+    - name: Install cargo-edit
+      run: cargo install cargo-edit
+
+    - name: Bump crate versions
+      run: cargo set-version --workspace ${{ steps.set-version.outputs.version }}
+
+    - name: Report differences for "prepare (release)" commit
+      run: git diff
+
+    - name: Commit cargo.toml and changelog
+      uses: stefanzweifel/git-auto-commit-action@v4
+      id: commit
+      with:
+        commit_message: Prepare ${{ steps.set-version.outputs.version }} release
+        commit_user_name: Adobe CAI Team
```

Comment:
Will this always just be Adobe CAI team, or can we identify the user doing the commit here?

Reply:
No, this is hard-coded. I don't _think_ I have a way to identify who invoked the workflow here.